### PR TITLE
ENH add run export

### LINKED ISSUE #59: OSX compatibility versions
So fftw 3.3.4 has this compatibility version

```
Load command 3
          cmd LC_ID_DYLIB
      cmdsize 56
         name @rpath/libfftw3f.3.dylib (offset 24)
   time stamp 1 Wed Dec 31 18:00:01 1969
      current version 8.4.0
compatibility version 8.0.0
```

However version 3.3.8 looks like this

```
Load command 3
          cmd LC_ID_DYLIB
      cmdsize 56
         name @rpath/libfftw3f.3.dylib (offset 24)
   time stamp 1 Wed Dec 31 18:00:01 1969
      current version 9.8.0
compatibility version 9.0.0
```

but the libraries themselves have file names like this

```
(base) localhost:lib Matt$ ls
libfftw3.3.dylib          libfftw3_threads.3.dylib  libfftw3f.3.dylib         libfftw3f_threads.3.dylib libfftw3l.3.dylib         libfftw3l_threads.3.dylib pkgconfig
libfftw3.a                libfftw3_threads.a        libfftw3f.a               libfftw3f_threads.a       libfftw3l.a               libfftw3l_threads.a
libfftw3.dylib            libfftw3_threads.dylib    libfftw3f.dylib           libfftw3f_threads.dylib   libfftw3l.dylib           libfftw3l_threads.dylib
libfftw3.la               libfftw3_threads.la       libfftw3f.la              libfftw3f_threads.la      libfftw3l.la              libfftw3l_threads.la
```

So my questions are 

1. Why aren't the dylibs named with their compat versions (e.g., libfftw3.8.dylib or whatever)?
2. As a practical issue, do we need to break up the build so that the dylibs can be properly versioned?

I suppose also we could pin fftw down to the micro version globally

### PR DESCRIPTION
**Pull request by beckermr**
Checklist
* [x] Used a [fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

closes #59

### CODE REVIEW

**Comment by beckermr** [comment]
@conda-forge-admin, please rerender

**Comment by beckermr** [comment]
We should remove the pin_run_as_build from the pinnings file once this merges

**Comment by jjhelmus** [comment]
LGTM, thanks @beckermr

File: recipe/meta.yaml [hunk 1]
```diff
@@ -33,8 +33,9 @@ build:
   {% endif %}
   string: "{{ mpi_prefix }}_h{{ PKG_HASH }}_{{ build }}"
 
-  {% if mpi != 'nompi' %}
   run_exports:
+    - {{ pin_compatible('fftw', max_pin='x') }}
```
Comment (isuruf) [inline recipe/meta.yaml]: Shouldn't this be `pin_subpackage`?

Comment (beckermr) [reply recipe/meta.yaml]: You know more than me. Probably?

Comment (beckermr) [reply recipe/meta.yaml]: It generated this set of pins when I rebuild galsim

```
astropy >=2.0, eigen >=3.3, fftw >=3.3.8,<4.0a0, future >=0.15, libcxx >=9.0.0, lsstdesc.coord >=1.0.5, numpy >=1.14.6,<2.0a0, pybind11 >=2.2, python >=2.7,<2.8.0a0
```

Comment (beckermr) [reply recipe/meta.yaml]: Those seem fine, but I might be missing something as usual. :)

Comment (beckermr) [reply recipe/meta.yaml]: That rebuild also has the pin_run_as_build stuff from a rerender, so not 100% clear which thing did what. Package level variants should have higher priority IIUIC.

**Pull request merged**